scripts/utils/generateMain.js

```diff
@@ -160,4 +164,70 @@ function isSnakeCased (str) {
   return !!~str.indexOf('_')
 }
 
+function toPascalCase (str) {
+  return str[0].toUpperCase() + str.slice(1)
+}
+
+function buildMethodDefinition (api, name, hasBody) {
+  const Name = toPascalCase(name)
+  if (hasBody) {
+    let methods = [
+      { key: `${api}<RequestBody = any, Response = any, Context = unknown>()`, val: `Promise<ApiResponse<Response, Context>>` },
+      { key: `${api}<RequestBody = any, Response = any, Context = unknown>(params: RequestParams.${Name}<RequestBody>)`, val: `Promise<ApiResponse<Response, Context>>` },
+      { key: `${api}<RequestBody = any, Response = any, Context = unknown>(params: RequestParams.${Name}<RequestBody>, options: TransportRequestOptions)`, val: `Promise<ApiResponse<Response, Context>>` },
+      { key: `${api}<RequestBody = any, Response = any, Context = unknown>(callback: callbackFn<Response, Context>)`, val: `TransportRequestCallback` },
+      { key: `${api}<RequestBody = any, Response = any, Context = unknown>(params: RequestParams.${Name}<RequestBody>, callback: callbackFn<Response, Context>)`, val: `TransportRequestCallback` },
+      { key: `${api}<RequestBody = any, Response = any, Context = unknown>(params: RequestParams.${Name}<RequestBody>, options: TransportRequestOptions, callback: callbackFn<Response, Context>)`, val: `TransportRequestCallback` }
+    ]
+    if (isSnakeCased(api)) {
+      methods = methods.concat([
+        { key: `${camelify(api)}<RequestBody = any, Response = any, Context = unknown>()`, val: `Promise<ApiResponse<Response, Context>>` },
+        { key: `${camelify(api)}<RequestBody = any, Response = any, Context = unknown>(params: RequestParams.${Name}<RequestBody>)`, val: `Promise<ApiResponse<Response, Context>>` },
+        { key: `${camelify(api)}<RequestBody = any, Response = any, Context = unknown>(params: RequestParams.${Name}<RequestBody>, options: TransportRequestOptions)`, val: `Promise<ApiResponse<Response, Context>>` },
+        { key: `${camelify(api)}<RequestBody = any, Response = any, Context = unknown>(callback: callbackFn<Response, Context>)`, val: `TransportRequestCallback` },
+        { key: `${camelify(api)}<RequestBody = any, Response = any, Context = unknown>(params: RequestParams.${Name}<RequestBody>, callback: callbackFn<Response, Context>)`, val: `TransportRequestCallback` },
+        { key: `${camelify(api)}<RequestBody = any, Response = any, Context = unknown>(params: RequestParams.${Name}<RequestBody>, options: TransportRequestOptions, callback: callbackFn<Response, Context>)`, val: `TransportRequestCallback` }
+      ])
+    }
+    return methods
+  } else {
+    let methods = [
+      { key: `${api}<Response = any, Context = unknown>()`, val: `Promise<ApiResponse<Response, Context>>` },
```

Comment:
why Response is `any`, Context is `unknown`?

Reply:
Because I'm trying to avoid any breaking change for current users. The current default value of generics is `any`, while it has never been set for the context, so I've decided to use `unknown`, which is more appropriate.

Reply:
I've just updated the code to improve the request body definition, it should only be `Record<string, any> | string | Buffer | ReadableStream`.
In the future, we should update the `Record<string, any>` with the real interface.

Reply:
I'm thinking of defining the `BodyType` as follows:
```ts
type BodyType<T = Record<string, any>> = T | string | Buffer | ReadableStream
```
In this way, we should be able to add the correct interfaces gradually.
I'm not sure if it's fine, but ts is not complaining :D